dataactbroker/handlers/fileHandler.py

```diff
@@ -831,27 +824,35 @@ def load_d_file(self, url, upload_name, timestamped_name, job_id, isLocal):
             sess.commit()
             raise e
 
-    def getRequestParamsForGenerate(self):
+    def get_request_params_for_generate(self, detached=False):
         """ Pull information out of request object and return it
 
         Returns: tuple of submission ID and file type
 
         """
-        requestDict = RequestDictionary(self.request)
-        if not (requestDict.exists("submission_id") and requestDict.exists("file_type")):
+        request_dict = RequestDictionary.derive(self.request)
+
+        if detached:
+            if not set(['file_type', 'cgac_code', 'start', 'end']).issubset(request_dict.keys()):
+                raise ResponseException("Generate detached file route for D files requires file_type, cgac_code, "
+                                        "start, and end",
+                                        StatusCode.CLIENT_ERROR)
+            return request_dict['file_type'], request_dict['cgac_code'], request_dict['start'], request_dict['end']
+
+        if not set(['file_type', 'submission_id']).issubset(request_dict.keys()):
             raise ResponseException("Generate file route requires submission_id and file_type",
                                     StatusCode.CLIENT_ERROR)
 
-        submission_id = requestDict.getValue("submission_id")
-        file_type = requestDict.getValue("file_type")
+        submission_id = request_dict['submission_id']
+        file_type = request_dict['file_type']
         return submission_id, file_type
```

Comment:
This function also returns multiple types of values. Can we standardize to a single return type?